services/provider/server/server.go

```diff
@@ -404,6 +404,28 @@ func (s *OCSProviderServer) GetDesiredClientState(ctx context.Context, req *pb.G
 			klog.Error(err)
 			return nil, status.Errorf(codes.Internal, "failed to produce client state hash")
 		}
+		consumerConfigMap := &corev1.ConfigMap{}
+		consumerConfigMap.Name = consumer.Status.ResourceNameMappingConfigMap.Name
+		consumerConfigMap.Namespace = consumer.Namespace
```

Comment:
We are also loading the consumer configMap inside the getKubeResources, maybe we should load it once outside and pass it around

Reply:
Moved consumerConfig generation to a helper function